engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -561,6 +584,48 @@ struct blinding_light_t : public paladin_spell_t
   }
 };
 
+// Auras ===============================================
+
+struct paladin_aura_base_t : public paladin_spell_t
+{
+  buff_t* aura_buff;
+  paladin_aura_base_t( const std::string& n, paladin_t* p, const spell_data_t* s ) :
+    paladin_spell_t( n, p, s )
+  {
+    harmful = false;
+    aura_buff = nullptr;
+  }
+
+  void execute() override
+  {
+    paladin_spell_t::execute();
+    assert( aura_buff != nullptr && "Paladin auras must have aura_buff set to their appropriate buff" );
```

Comment:
You could move this assert to paladin_spell_t::init_finished() instead of running it every execute()